### PR TITLE
Various improvements for Streamlit demo

### PR DESCRIPTION
**Pull request by jaceksan**
Extend model containers with missing properties.
Allow using model_container without embedded validObjects function, so it can be cached(pickled) by Streamlit

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/gooddata_sdk/catalog/workspace/model_container.py [hunk 1]
```diff
@@ -97,36 +117,49 @@ def find_label_attribute(self, id_obj: IdObjType) -> Union[CatalogAttribute, Non
         return None
 
     def _valid_objects(self, ctx: ValidObjectsInputType) -> ValidObjects:
-        return self._valid_obf_fun(ctx)
+        if self._valid_obj_fun:
+            return self._valid_obj_fun(ctx)
+        else:
+            return {}
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/catalog/workspace/model_container.py:123]: Return type should be ValidObjects and you return empty dictionary. Is it intended?

File: gooddata-sdk/gooddata_sdk/insight.py [hunk 2]
```diff
@@ -264,12 +264,25 @@ def item(self) -> Optional[dict[str, Any]]:
 
         return None
 
+    @property
+    def aggregation(self) -> Optional[str]:
+        if "measureDefinition" in self._d:
+            return self._d["measureDefinition"].get("aggregation")
+
+        return None
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/insight.py:272]: Simplification:

`return self._d.get("measureDefinition",{}).get("aggregation")`

or using safeget .

File: gooddata-sdk/gooddata_sdk/insight.py [hunk 3]
```diff
@@ -264,12 +264,25 @@ def item(self) -> Optional[dict[str, Any]]:
...
     def item_id(self) -> Optional[str]:
         item = self.item
 
         return item["identifier"]["id"] if item is not None else None
 
+    @property
+    def item_type(self) -> Optional[str]:
+        item = self.item
+
+        return item["identifier"]["type"] if item is not None else None
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/insight.py:284]: Again we can use safeget here.

File: gooddata-sdk/gooddata_sdk/catalog/workspace/model_container.py [hunk 4]
```diff
@@ -97,36 +117,49 @@ def find_label_attribute(self, id_obj: IdObjType) -> Union[CatalogAttribute, Non
...
+        new_metrics = self.metrics
+        if self._valid_obj_fun:
+            valid_objects = self._valid_objects(ctx)
+            new_datasets, new_metrics = self.filter_by_valid_objects(valid_objects)
         return CatalogWorkspaceContent(
-            self._valid_obf_fun,
+            self._valid_obj_fun,
             datasets=new_datasets,
             metrics=new_metrics,
         )
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/catalog/workspace/model_container.py:157]:
```python
if not self._valid_obj_fun:
    # or alternatively use attr.evolve(self) to make a copy
    return self

valid_objects = self._valid_objects(ctx)
new_datasets, new_metrics = self.filter_by_valid_objects(valid_objects)
return CatalogWorkspaceContent(
            self._valid_obj_fun,
            datasets=new_datasets,
            metrics=new_metrics,
        )
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by hkad98 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR268

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**